// sanity-plugin-dashboard-widget-shopify/lib/shopify-intro/app.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _dashboard = require("@sanity/dashboard");

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _user = _interopRequireDefault(require("part:@sanity/base/user"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ResourceGuide = _interopRequireDefault(require("./components/ResourceGuide"));

var _ResourceScreencast = _interopRequireDefault(require("./components/ResourceScreencast"));

var _templateObject;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Container = (0, _styledComponents["default"])(_ui.Flex)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  flex-direction: column;\n\n  [data-eq-min~='0'] > & {\n  }\n\n  [data-eq-min~='1'] > & {\n    flex-direction: row;\n  }\n"])));

function Widget() {
  var _useState = (0, _react.useState)(),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      displayName = _useState2[0],
      setDisplayName = _useState2[1];

  (0, _react.useEffect)(function () {
    function getUser() {
      return _getUser.apply(this, arguments);
    }

    function _getUser() {
      _getUser = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
        var _yield$userStore$getU, displayName;

        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.next = 2;
                return _user["default"].getUser('me');

              case 2:
                _yield$userStore$getU = _context.sent;
                displayName = _yield$userStore$getU.displayName;
                setDisplayName(displayName);

              case 5:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));
      return _getUser.apply(this, arguments);
    }

    getUser();
  }, []);
  return /*#__PURE__*/_react["default"].createElement(_dashboard.DashboardWidget, {
    header: displayName ? "Hello, ".concat(displayName, "!") : "Welcome!"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Card, null, /*#__PURE__*/_react["default"].createElement(_ui.ElementQuery, null, /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    flex: 1,
    padding: [4, 4, 5]
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 4
  }, /*#__PURE__*/_react["default"].createElement("img", {
    src: "https://cdn.sanity.io/images/3do82whm/next/2c15d260780e68fbf7db19d5580b011f266fc839-1982x1228.png?w=1200",
    style: {
      width: '100%'
    }
  }), /*#__PURE__*/_react["default"].createElement(_ui.Text, null, "Thanks for installing Sanity Connect for Shopify. Welcome to your very own Sanity Studio!"), /*#__PURE__*/_react["default"].createElement(_ui.Text, null, "This is where you\u2019ll see your products and variants that have been synced from Shopify. It\u2019s also where you can add metadata to enrich your product information with fields for image galleries, block content for detailed product descriptions, SEO that lets you optimize your search engine results page (SERP) listings, and more."), /*#__PURE__*/_react["default"].createElement(_ui.Text, null, "Need another content model to support your editorial needs? With Sanity, you can update and change your content model as needed to support even the most advanced requirements."), /*#__PURE__*/_react["default"].createElement(_ui.Text, null, "Have a look around at the resources available to you here, and do join the #shopify channel in our", ' ', /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://slack.sanity.io/",
    rel: "noopener noreferrer",
    target: "_blank"
  }, "Slack community"), ' ', "to get help with any questions you might have.")), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 6
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 4
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://www.sanity.io/docs/sanity-connect-for-shopify",
    target: "_blank",
    rel: "noreferrer"
  }, /*#__PURE__*/_react["default"].createElement(_icons.RocketIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Documentation")), /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://www.sanity.io/docs/reference",
    target: "_blank",
    rel: "noreferrer"
  }, /*#__PURE__*/_react["default"].createElement(_icons.BookIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Reference docs")), /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://slack.sanity.io/",
    target: "_blank",
    rel: "noreferrer"
  }, /*#__PURE__*/_react["default"].createElement(_icons.CommentIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Community support"))))), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    flex: 1,
    padding: [0, 0, 4]
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, null, /*#__PURE__*/_react["default"].createElement(_ResourceScreencast["default"], {
    title: "Getting started with Sanity Connect for Shopify",
    youtubeId: "TUY84_V5f9Q"
  }), /*#__PURE__*/_react["default"].createElement(_ResourceGuide["default"], {
    href: "https://www.sanity.io/guides/structured-content-patterns-for-e-commerce",
    posterUrl: "https://cdn.sanity.io/images/3do82whm/next/ff7dcdd8ce4d5039334db85f7dbbbc8c35d534ee-1200x900.png?rect=0,75,1150,720&w=600",
    title: "Best practices for e-commerce"
  })))))));
}

var _default = Widget;
exports["default"] = _default;
