// @sanity/dashboard/lib/widgets/sanityTutorials/SanityTutorials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _get2 = _interopRequireDefault(require("lodash/get"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ui = require("@sanity/ui");

var _DashboardTool = require("../../DashboardTool");

var _Tutorial = _interopRequireDefault(require("./Tutorial"));

var _dataAdapter = _interopRequireDefault(require("./dataAdapter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var urlBuilder = _dataAdapter.default.urlBuilder,
    getFeed = _dataAdapter.default.getFeed;

function createUrl(slug, type) {
  if (type === 'tutorial') {
    return "https://www.sanity.io/docs/tutorials/".concat(slug.current);
  } else if (type === 'guide') {
    return "https://www.sanity.io/docs/guides/".concat(slug.current);
  }

  return false;
}

class SanityTutorials extends _react.default.Component {
  constructor() {
    super(...arguments);

    _defineProperty(this, "state", {
      feedItems: []
    });
  }

  componentDidMount() {
    var templateRepoId = this.props.templateRepoId;
    this.subscription = getFeed(templateRepoId).subscribe(response => {
      this.setState({
        feedItems: response.items
      });
    });
  }

  componentWillUnmount() {
    if (this.subscription) {
      this.subscription.unsubscribe();
    }
  }

  render() {
    var feedItems = this.state.feedItems;
    var title = 'Learn about Sanity';
    return /*#__PURE__*/_react.default.createElement(_DashboardTool.DashboardWidget, {
      header: title
    }, /*#__PURE__*/_react.default.createElement(_ui.Flex, {
      as: "ul",
      overflow: "auto",
      align: "stretch",
      paddingY: 2
    }, feedItems === null || feedItems === void 0 ? void 0 : feedItems.map((feedItem, index) => {
      if (!feedItem.title || !feedItem.guideOrTutorial && !feedItem.externalLink) {
        return null;
      }

      var presenter = feedItem.presenter || (0, _get2.default)(feedItem, 'guideOrTutorial.presenter') || {};
      var subtitle = (0, _get2.default)(feedItem, 'category');
      var _feedItem$guideOrTuto = feedItem.guideOrTutorial,
          guideOrTutorial = _feedItem$guideOrTuto === void 0 ? {} : _feedItem$guideOrTuto;
      return /*#__PURE__*/_react.default.createElement(_ui.Flex, {
        as: "li",
        key: feedItem._id,
        paddingRight: index < (feedItems === null || feedItems === void 0 ? void 0 : feedItems.length) - 1 ? 1 : 3,
        paddingLeft: index === 0 ? 3 : 0,
        align: "stretch",
        flex: "0 0 27.5%",
        style: {
          minWidth: 272,
          width: '30%'
        }
      }, /*#__PURE__*/_react.default.createElement(_Tutorial.default, {
        title: feedItem.title,
        href: createUrl(guideOrTutorial.slug, guideOrTutorial._type) || feedItem.externalLink,
        presenterName: presenter.name,
        presenterSubtitle: subtitle,
        showPlayIcon: feedItem.hasVideo,
        posterURL: urlBuilder.image(feedItem.poster).height(360).url()
      }));
    })));
  }

}

_defineProperty(SanityTutorials, "propTypes", {
  templateRepoId: _propTypes.default.string
});

_defineProperty(SanityTutorials, "defaultProps", {
  templateRepoId: null
});

var _default = SanityTutorials;
exports.default = _default;
