// @sanity/dashboard/lib/widgets/sanityTutorials/Tutorial.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ui = require("@sanity/ui");

var _icons = require("@sanity/icons");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _templateObject, _templateObject2, _templateObject3, _templateObject4;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var PlayIconBox = (0, _styledComponents.default)(_ui.Box)(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  position: absolute;\n  top: 50%;\n  left: 50%;\n  transform: translate(-50%, -50%);\n\n  &:before {\n    content: '';\n    position: absolute;\n    top: 50%;\n    left: 50%;\n    transform: translate(-50%, -50%);\n    width: 2.75em;\n    height: 2.75em;\n    border-radius: 50%;\n    background: ", ";\n    opacity: 0.75;\n  }\n"])), _ref => {
  var theme = _ref.theme;
  return theme.sanity.color.card.enabled.bg;
});
var Root = (0, _styledComponents.default)(_ui.Flex)(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  &:hover {\n    ", " {\n      &:before {\n        opacity: 1;\n      }\n    }\n  }\n"])), PlayIconBox);
var PosterCard = (0, _styledComponents.default)(_ui.Card)(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  width: 100%;\n  padding-bottom: calc(9 / 16 * 100%);\n  position: relative;\n"])));

var Poster = _styledComponents.default.img(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  height: 100%;\n  width: 100%;\n  object-fit: cover;\n  display: block;\n\n  &:not([src]) {\n    display: none;\n  }\n"])));

class Tutorial extends _react.default.PureComponent {
  render() {
    var _this$props = this.props,
        title = _this$props.title,
        posterURL = _this$props.posterURL,
        showPlayIcon = _this$props.showPlayIcon,
        href = _this$props.href,
        presenterName = _this$props.presenterName,
        presenterSubtitle = _this$props.presenterSubtitle;
    return /*#__PURE__*/_react.default.createElement(Root, {
      flex: 1
    }, /*#__PURE__*/_react.default.createElement(_ui.Card, {
      sizing: "border",
      flex: 1,
      padding: 2,
      radius: 2,
      as: "a",
      href: href,
      target: "_blank",
      rel: "noopener noreferrer",
      style: {
        position: 'relative'
      }
    }, /*#__PURE__*/_react.default.createElement(_ui.Flex, {
      direction: "column",
      style: {
        height: '100%'
      }
    }, posterURL && /*#__PURE__*/_react.default.createElement(PosterCard, {
      marginBottom: 1
    }, /*#__PURE__*/_react.default.createElement(Poster, {
      src: posterURL
    }), showPlayIcon && /*#__PURE__*/_react.default.createElement(PlayIconBox, {
      display: "flex"
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, {
      align: "center"
    }, /*#__PURE__*/_react.default.createElement(_icons.PlayIcon, null)))), /*#__PURE__*/_react.default.createElement(_ui.Flex, {
      direction: "column",
      justify: "space-between",
      paddingY: 2,
      flex: 1
    }, /*#__PURE__*/_react.default.createElement(_ui.Heading, {
      as: "h3",
      size: 1
    }, title), /*#__PURE__*/_react.default.createElement(_ui.Box, {
      marginTop: 4
    }, /*#__PURE__*/_react.default.createElement(_ui.Stack, {
      space: 2,
      flex: 1
    }, /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 1
    }, presenterName), /*#__PURE__*/_react.default.createElement(_ui.Text, {
      size: 0,
      style: {
        opacity: 0.7
      }
    }, presenterSubtitle)))))));
  }

}

_defineProperty(Tutorial, "propTypes", {
  title: _propTypes.default.string.isRequired,
  posterURL: _propTypes.default.string,
  href: _propTypes.default.string.isRequired,
  showPlayIcon: _propTypes.default.bool,
  presenterName: _propTypes.default.string.isRequired,
  presenterSubtitle: _propTypes.default.string.isRequired
});

_defineProperty(Tutorial, "defaultProps", {
  posterURL: null,
  showPlayIcon: false
});

var _default = Tutorial;
exports.default = _default;
