// sanity-plugin-dashboard-widget-shopify/lib/shopify-intro/components/ResourceGuide.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _color = require("@sanity/color");

var _ui = require("@sanity/ui");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _templateObject, _templateObject2, _templateObject3;

var Container = (0, _styledComponents["default"])(_ui.Card)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  background: ", ";\n  padding-bottom: calc(9 / 16 * 100%);\n  position: relative;\n  width: 100%;\n"])), _color.hues.gray[100].hex);
var IconContainer = (0, _styledComponents["default"])(_ui.Box)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 50%;\n  left: 50%;\n  transform: translate(-50%, -50%);\n\n  &:before {\n    background: ", ";\n    border-radius: 50%;\n    content: '';\n    height: 3.25em;\n    opacity: 0.9;\n    position: absolute;\n    left: 50%;\n    top: 50%;\n    transform: translate(-50%, -50%);\n    width: 3.25em;\n  }\n"])), function (_ref) {
  var theme = _ref.theme;
  return theme.sanity.color.card.enabled.bg;
});

var Poster = _styledComponents["default"].img(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  height: 100%;\n  width: 100%;\n  object-fit: cover;\n  display: block;\n"])));

var Resource = function Resource(props) {
  var byline = props.byline,
      href = props.href,
      posterUrl = props.posterUrl,
      title = props.title;
  return /*#__PURE__*/_react["default"].createElement(_ui.Card, {
    as: "a",
    href: href,
    rel: "noopener noreferrer",
    padding: 4,
    target: "_blank"
  }, /*#__PURE__*/_react["default"].createElement(Container, {
    overflow: "hidden",
    radius: 1
  }, posterUrl && /*#__PURE__*/_react["default"].createElement(Poster, {
    src: posterUrl
  })), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    as: "h3",
    size: 2,
    weight: "semibold"
  }, title)), byline && /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    muted: true,
    size: 1
  }, byline)), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 4
  }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
    muted: true,
    size: 0
  }, "Guide")));
};

var _default = Resource;
exports["default"] = _default;
