// sanity-plugin-dashboard-widget-shopify/lib/shopify-intro/components/ResourceScreencast.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _ui = require("@sanity/ui");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _templateObject, _templateObject2;

var Container = (0, _styledComponents["default"])(_ui.Card)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  padding-bottom: calc(9 / 16 * 100%);\n  position: relative;\n  width: 100%;\n"])));

var Iframe = _styledComponents["default"].iframe(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  height: 100%;\n  width: 100%;\n  object-fit: cover;\n  display: block;\n"])));

var Resource = function Resource(props) {
  var byline = props.byline,
      title = props.title,
      youtubeId = props.youtubeId;
  var href = "https://www.youtube.com/watch?v=".concat(youtubeId);
  return /*#__PURE__*/_react["default"].createElement(_ui.Card, {
    as: "a",
    href: href,
    rel: "noopener noreferrer",
    padding: 4,
    target: "_blank"
  }, /*#__PURE__*/_react["default"].createElement(Container, {
    overflow: "hidden",
    radius: 1
  }, youtubeId && /*#__PURE__*/_react["default"].createElement(Iframe, {
    frameBorder: "0",
    src: "https://www.youtube.com/embed/".concat(youtubeId, "?modestbranding=1&showinfo=0&rel=0&iv_load_policy=3")
  })), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    as: "h3",
    size: 2,
    weight: "semibold"
  }, title)), byline && /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    muted: true,
    size: 1
  }, byline)), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 4
  }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
    muted: true,
    size: 0
  }, "Screencast")));
};

var _default = Resource;
exports["default"] = _default;
