// sanity-plugin-dashboard-widget-shopify/lib/shopify-connect/components/PanelShopify.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _client = require("../../lib/client");

var _Row = _interopRequireDefault(require("./Row"));

var clientConfig = _client.sanityClient.config();

var PanelShopify = function PanelShopify(props) {
  var _syncStatus$store;

  var syncStatus = props.syncStatus;
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    flex: 1
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
    muted: true,
    size: 1
  }, "Shopify store"), syncStatus === null ? /*#__PURE__*/_react["default"].createElement(_ui.Card, {
    marginTop: 2,
    padding: 4,
    tone: "caution"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, null, /*#__PURE__*/_react["default"].createElement(_ui.Box, null, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, /*#__PURE__*/_react["default"].createElement(_icons.WarningOutlineIcon, null))), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    flex: 1,
    marginLeft: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 4
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "semibold"
  }, "Widget configuration is missing"), /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1
  }, "Install the official Sanity Connect app in the Shopify app store and connect it to this project (", /*#__PURE__*/_react["default"].createElement("code", null, clientConfig === null || clientConfig === void 0 ? void 0 : clientConfig.projectId), ") to enable this widget."), /*#__PURE__*/_react["default"].createElement(_ui.Box, null, /*#__PURE__*/_react["default"].createElement(_ui.Button, {
    as: "a",
    fontSize: 1,
    href: "https://apps.shopify.com/",
    mode: "ghost",
    rel: "noopener noreferrer",
    target: "_blank",
    text: "Install Sanity Connect",
    tone: "primary"
  })))))) : /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    title: "store ID",
    value: (syncStatus === null || syncStatus === void 0 ? void 0 : (_syncStatus$store = syncStatus.store) === null || _syncStatus$store === void 0 ? void 0 : _syncStatus$store.replace('.myshopify.com', '')) || '-'
  })), syncStatus !== null && /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 5
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: (syncStatus === null || syncStatus === void 0 ? void 0 : syncStatus.store) && "https://".concat(syncStatus === null || syncStatus === void 0 ? void 0 : syncStatus.store, "/admin"),
    rel: "noopener noreferrer",
    style: {
      opacity: syncStatus !== null && syncStatus !== void 0 && syncStatus.store ? 1 : 0.5,
      pointerEvents: syncStatus !== null && syncStatus !== void 0 && syncStatus.store ? 'auto' : 'none'
    },
    target: "_blank"
  }, /*#__PURE__*/_react["default"].createElement(_icons.LaunchIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Shopify store admin"))));
};

var _default = PanelShopify;
exports["default"] = _default;
