// sanity-plugin-dashboard-widget-shopify/lib/shopify-connect/components/PanelUpdatedProducts.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _SyncItem = _interopRequireDefault(require("./SyncItem"));

var PanelUpdatedProducts = function PanelUpdatedProducts(props) {
  var updates = props.updates;
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, null, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
    muted: true,
    size: 1
  }, "Recently updated products"), /*#__PURE__*/_react["default"].createElement(_ui.Box, null, updates && updates.length > 0 && (updates === null || updates === void 0 ? void 0 : updates.map(function (update) {
    return /*#__PURE__*/_react["default"].createElement(_SyncItem["default"], {
      documentId: update.documentId,
      error: update.error,
      key: "".concat(update.documentId, "-").concat(update.timestamp),
      timestamp: update.timestamp,
      type: update.type
    });
  })))));
};

var _default = PanelUpdatedProducts;
exports["default"] = _default;
