// sanity-plugin-dashboard-widget-shopify/lib/shopify-connect/components/PanelSanity.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _client = require("../../lib/client");

var _Row = _interopRequireDefault(require("./Row"));

var clientConfig = _client.sanityClient.config();

var PanelSanity = function PanelSanity() {
  return /*#__PURE__*/_react["default"].createElement(_ui.Box, null, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 3
  }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
    muted: true,
    size: 1
  }, "Sanity project"), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    title: "project ID",
    value: clientConfig.projectId
  }), /*#__PURE__*/_react["default"].createElement(_Row["default"], {
    title: "dataset",
    value: clientConfig.dataset
  })), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 5
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: "https://manage.sanity.io/projects/".concat(clientConfig.projectId),
    rel: "noopener noreferrer",
    target: "_blank"
  }, /*#__PURE__*/_react["default"].createElement(_icons.LaunchIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Manage Sanity project"))));
};

var _default = PanelSanity;
exports["default"] = _default;
