// sanity-plugin-dashboard-widget-shopify/lib/shopify-connect/components/SyncItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _color = require("@sanity/color");

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _router = require("part:@sanity/base/router");

var _react = _interopRequireWildcard(require("react"));

var _reactTimeAgo = _interopRequireDefault(require("react-time-ago"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _client = require("../../lib/client");

var _templateObject, _templateObject2, _templateObject3;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var TIME_OFFSET = 2500; // ms

var ImageContainer = (0, _styledComponents["default"])(_ui.Box)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  background: ", ";\n  height: 2.5em;\n  position: relative;\n  width: 2.5em;\n"])), _color.hues.gray[100].hex);

var PreviewImage = _styledComponents["default"].img(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  border-radius: 2px;\n  height: 100%;\n  object-fit: cover;\n  width: 100%;\n"])));

var TextPlaceholder = (0, _styledComponents["default"])(_ui.Card)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n  /* background: red; */\n  background: ", ";\n"])), function (props) {
  return props.visible ? 'none' : _color.hues.gray[50].hex;
});
TextPlaceholder.flex = 1;
TextPlaceholder.radius = 2;
var QUERY = "*[_id == $id][0]";
var PLACEHOLDER_DELAY = 1500; // ms

var TYPE_MAP = {
  create: 'Created',
  "delete": 'Deleted',
  update: 'Updated'
};

var SyncItem = function SyncItem(props) {
  var timestamp = props.timestamp,
      documentId = props.documentId,
      error = props.error,
      type = props.type;

  var _useState = (0, _react.useState)(true),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      imageVisible = _useState2[0],
      setImageVisible = _useState2[1];

  var _useState3 = (0, _react.useState)('loading'),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      mode = _useState4[0],
      setMode = _useState4[1];

  var _useState5 = (0, _react.useState)(),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      product = _useState6[0],
      setProduct = _useState6[1];

  var refTimeout = (0, _react.useRef)();

  var handleDocumentUpdate = function handleDocumentUpdate(document) {
    setProduct(document);
  }; // Hide image on error / 404


  var handleImageError = function handleImageError() {
    return setImageVisible(false);
  };

  (0, _react.useEffect)(function () {
    // Fetch product and listen to updates
    _client.sanityClient.fetch(QUERY, {
      id: documentId
    }).then(handleDocumentUpdate);

    var subscription = _client.sanityClient.listen(QUERY, {
      id: documentId
    }).subscribe(function (mutation) {
      if (mutation.transition === 'disappear') {
        handleDocumentUpdate(null);
      }

      if (mutation !== null && mutation !== void 0 && mutation.result) {
        handleDocumentUpdate(mutation.result);
      }
    }); // Clean up listeners on unmount


    return function () {
      subscription === null || subscription === void 0 ? void 0 : subscription.unsubscribe();
    };
  }, []);
  (0, _react.useEffect)(function () {
    // Wait one second before displaying
    refTimeout.current = setTimeout(function () {
      if (product === null) {
        setMode('inaccessible');
      }

      if (product) {
        setMode('ready');
      }
    }, PLACEHOLDER_DELAY);
    return function () {
      if (refTimeout.current) {
        clearTimeout(refTimeout.current);
      }
    };
  }, [product]);
  var isInaccessible = mode === 'inaccessible';
  var isLoading = mode === 'loading';
  var isReady = mode === 'ready';

  var Content = function Content() {
    var _product$store, _product$store2;

    return /*#__PURE__*/_react["default"].createElement(_ui.Card, {
      paddingY: 2
    }, /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
      align: "center"
    }, /*#__PURE__*/_react["default"].createElement(ImageContainer, null, isReady && (product !== null && product !== void 0 && (_product$store = product.store) !== null && _product$store !== void 0 && _product$store.previewImageUrl && imageVisible ? /*#__PURE__*/_react["default"].createElement(PreviewImage, {
      onError: handleImageError,
      src: product.store.previewImageUrl
    }) : /*#__PURE__*/_react["default"].createElement(_icons.ImageIcon, {
      style: {
        color: _color.hues.gray[700].hex,
        height: '100%',
        position: 'absolute',
        width: '100%'
      }
    }))), /*#__PURE__*/_react["default"].createElement(_ui.Box, {
      flex: 1,
      marginX: 3
    }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
      space: 2
    }, /*#__PURE__*/_react["default"].createElement(TextPlaceholder, {
      visible: !isLoading
    }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
      size: 2,
      textOverflow: "ellipsis"
    }, isInaccessible && /*#__PURE__*/_react["default"].createElement("em", null, "Document not found"), isReady && (product === null || product === void 0 ? void 0 : (_product$store2 = product.store) === null || _product$store2 === void 0 ? void 0 : _product$store2.title), "\xA0")), /*#__PURE__*/_react["default"].createElement(TextPlaceholder, {
      visible: !isLoading
    }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
      muted: true,
      size: 1,
      textOverflow: "ellipsis"
    }, isReady && timestamp && /*#__PURE__*/_react["default"].createElement(_reactTimeAgo["default"], {
      date: new Date(timestamp).getTime() - TIME_OFFSET,
      timeStyle: "round"
    }), isInaccessible && /*#__PURE__*/_react["default"].createElement("em", null, documentId), "\xA0")))), /*#__PURE__*/_react["default"].createElement(_ui.Flex, {
      align: "center",
      justify: "flex-end",
      style: {
        // background: 'red',
        flexShrink: 0,
        marginLeft: 'auto',
        width: '5em'
      }
    }, !isLoading && error && /*#__PURE__*/_react["default"].createElement(_ui.Tooltip, {
      content: /*#__PURE__*/_react["default"].createElement(_ui.Box, {
        padding: 2
      }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
        muted: true,
        size: 1
      }, error)),
      portal: true
    }, /*#__PURE__*/_react["default"].createElement(_ui.Box, {
      marginRight: 3
    }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
      size: 1,
      style: {
        color: _color.hues.red[500].hex
      }
    }, /*#__PURE__*/_react["default"].createElement(_icons.ErrorOutlineIcon, null)))), /*#__PURE__*/_react["default"].createElement(TextPlaceholder, {
      flex: 1,
      visible: !isLoading
    }, /*#__PURE__*/_react["default"].createElement(_ui.Label, {
      align: "right",
      muted: true,
      size: 0
    }, !isLoading && TYPE_MAP[type], "\xA0")))));
  };

  if (!product) {
    return /*#__PURE__*/_react["default"].createElement(Content, null);
  }

  return /*#__PURE__*/_react["default"].createElement(_router.IntentLink, {
    as: "a",
    intent: "edit",
    params: {
      id: documentId
    },
    style: {
      textDecoration: 'none'
    }
  }, /*#__PURE__*/_react["default"].createElement(Content, null));
};

var _default = SyncItem;
exports["default"] = _default;
