// sanity-plugin-dashboard-widget-shopify/lib/shopify-connect/app.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _color = require("@sanity/color");

var _dashboard = require("@sanity/dashboard");

var _icons = require("@sanity/icons");

var _ui = require("@sanity/ui");

var _router = require("part:@sanity/base/router");

var _react = _interopRequireWildcard(require("react"));

var _client = require("../lib/client");

var _PanelManualSync = _interopRequireDefault(require("./components/PanelManualSync"));

var _PanelSanity = _interopRequireDefault(require("./components/PanelSanity"));

var _PanelShopify = _interopRequireDefault(require("./components/PanelShopify"));

var _PanelUpdatedProducts = _interopRequireDefault(require("./components/PanelUpdatedProducts"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

// import { useRouter } from '@sanity/base/router'
var QUERY = "*[_type == \"sanity.shopify.sync\"] | order(_updatedAt desc) [0]";

function Widget() {
  var _syncStatus$log;

  var _useState = (0, _react.useState)(),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      syncStatus = _useState2[0],
      setSyncStatus = _useState2[1];

  var router = (0, _router.useRouter)();

  var handleSyncUpdate = function handleSyncUpdate(document) {
    setSyncStatus(document);
  };

  (0, _react.useEffect)(function () {
    // Fetch sync status and listen to updates
    _client.sanityClient.fetch(QUERY).then(handleSyncUpdate);

    var subscription = _client.sanityClient.listen(QUERY).subscribe(function (mutation) {
      if (mutation.transition === 'disappear') {
        handleSyncUpdate(null);
      }

      if (mutation !== null && mutation !== void 0 && mutation.result) {
        handleSyncUpdate(mutation.result);
      }
    }); // Clean up listeners on unmount


    return function () {
      subscription === null || subscription === void 0 ? void 0 : subscription.unsubscribe();
    };
  }, []);

  var Divider = function Divider() {
    return /*#__PURE__*/_react["default"].createElement(_ui.Box, {
      style: {
        background: _color.hues.gray[200].hex,
        height: '1px',
        width: '100%'
      }
    });
  };

  var updatedProducts = syncStatus === null || syncStatus === void 0 ? void 0 : (_syncStatus$log = syncStatus.log) === null || _syncStatus$log === void 0 ? void 0 : _syncStatus$log.slice(0, 5);
  return /*#__PURE__*/_react["default"].createElement(_dashboard.DashboardWidget, {
    header: "Shopify Connect"
  }, /*#__PURE__*/_react["default"].createElement(_ui.Card, {
    padding: 4
  }, /*#__PURE__*/_react["default"].createElement(_ui.Stack, {
    space: 4
  }, /*#__PURE__*/_react["default"].createElement(_PanelShopify["default"], {
    syncStatus: syncStatus
  }), /*#__PURE__*/_react["default"].createElement(Divider, null), /*#__PURE__*/_react["default"].createElement(_PanelSanity["default"], null), /*#__PURE__*/_react["default"].createElement(Divider, null), (syncStatus === null || syncStatus === void 0 ? void 0 : syncStatus.store) && /*#__PURE__*/_react["default"].createElement(_PanelUpdatedProducts["default"], {
    updates: updatedProducts
  }), /*#__PURE__*/_react["default"].createElement(_PanelManualSync["default"], {
    status: syncStatus === null || syncStatus === void 0 ? void 0 : syncStatus.status
  }), updatedProducts && /*#__PURE__*/_react["default"].createElement(_ui.Box, {
    marginTop: 2
  }, /*#__PURE__*/_react["default"].createElement(_ui.Text, {
    size: 1,
    weight: "medium"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    href: "#",
    onClick: function onClick() {
      // Navigate to products root
      router.navigateUrl('/desk/products');
    }
  }, /*#__PURE__*/_react["default"].createElement(_icons.MasterDetailIcon, {
    style: {
      marginRight: '0.1em'
    }
  }), "Show all products"))))));
}

var _default = Widget;
exports["default"] = _default;
